lib/Analysis/NoiseAnalysis/BGV/NoiseCanEmbModel.h

```diff
@@ -0,0 +1,74 @@
+#ifndef INCLUDE_ANALYSIS_NOISEANALYSIS_BGV_NOISECANEMBMODEL_H_
+#define INCLUDE_ANALYSIS_NOISEANALYSIS_BGV_NOISECANEMBMODEL_H_
+
+#include <cassert>
+#include <string>
+
+#include "lib/Analysis/NoiseAnalysis/Noise.h"
+#include "lib/Parameters/BGV/Params.h"
+
+namespace mlir {
+namespace heir {
+namespace bgv {
+
+// coefficient embedding noise model
+// both average-case (from HPS19/KPZ21) and worst-case
+// use template here just for the sake of code reuse
+// W for worst-case, P for public key
+template <bool P>
+class NoiseCanEmbModel {
+ public:
+  // for KPZ21, NoiseState stores the bound ||e|| for error e.
```

Comment:
```suggestion
  // for MMGLA22, NoiseState stores the bound `||m + t * e||^{can}` for error e.
```

Reply:
Sorry, for the obvious miss. Corrected the comment